packages/database/src/models/User.ts

```diff
@@ -54,7 +59,12 @@ export class User extends Model {
     const { password, ...otherValues } = values;
     if (!password) return values;
 
-    return { ...otherValues, password: await User.hashPassword(password) };
+    // Only hash if the password is not already hashed (to avoid rehashing when syncing)
```

Comment:
Yes, BUT we must forbid passwords that start with the hash signature, otherwise we're open to someone setting a password that starts with a bcrypt prefix and never hashing it.

Reply:
Why was rehashing on sync not an issue before @khoi-bes?